cpp/src/structs/utilities.hpp

```diff
@@ -123,6 +125,29 @@ void superimpose_parent_nulls(bitmask_type const* parent_null_mask,
                               rmm::cuda_stream_view stream,
                               rmm::mr::device_memory_resource* mr);
 
+/**
+ * @brief Pushdown nulls from a parent mask into a child column, using bitwise AND.
+ *
+ * Rather than modify the argument column, this function constructs new equivalent column_view
+ * instances, with new null mask values. This function returns both a (possibly new) column,
+ * and the device_buffer instances to support any modified null masks.
+ * 1. If the specified column is not STRUCT, the column is returned unmodified, with no new
+ *    supporting device_buffer instances.
+ * 2. If the column is STRUCT, the null masks of the parent and child are bitwise-ANDed, and a
+ *    modified column_view is returned. This applies recursively to support
+ *
+ * @param parent The parent (possibly STRUCT) column whose nulls need to be pushed to its members.
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr     Device memory resource used to allocate new device memory.
+ * @return A pair of:
+ *         1. column_view with nulls pushed down to child columns, as appropriate.
+ *         2. Supporting device_buffer instances, for any newly constructed null masks.
+ */
+std::tuple<cudf::column_view, std::vector<rmm::device_buffer>> superimpose_parent_nulls(
```

Comment:
I wonder why do we need to take out the "supporting device_buffers`? Why don't just leave them to the newly constructed `column_view` instances to own? We can construct a `column_view` from a free device buffer, which will take ownership of that buffer.

Reply:
I'm not sure I understand the suggestion. Please rephrase? How would a `column_view` own a `device_buffer`?

Reply:
Sorry I was wrong---`column_view` can't own bitmask.

Reply:
Phew. I thought I missed something.

The previous function (`superimpose_parent_nulls()`) does as you suggest, since it modifies in place.